nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -179,62 +169,95 @@ class WeightNodeParams:
     compression_config = WeightCompressionConfig()
 
 
-def _int8_compress(
-    weight: np.ndarray, reduction_axes: Union[int, Tuple[int]]
+def _do_integer_quantization(
+    weight: np.ndarray, reduction_axes: Union[int, Tuple[int]], config: WeightCompressionConfig
 ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
     """
-    Do unsigned int8 asymmetric weight compression - quantization to [0, 255] range.
+    The method quantizes the given weights to integer data type in accordance with the compression config.
+    The config defines a quantization mode:
+        INT8 mode refers to unsigned int8 asymmetric weight compression - quantization to [0, 255] range.
+        INT4_ASYM - to unsigned int4 asymmetric weight compression - quantization to [0, 15] range.
+        INT4_SYM - to unsigned int4 symmetric weight compression - quantization to [0, 15] range.
+        NF4 mode requires a dedicated procedure and it is not supported in this method.
+    One of the parameter of compression config is a group size. Quantization is per-channel, if group size equals to -1,
+    otherwise it's per-group, i.e. group size number of weights in the channel dimension share quantization parameters
+    (scales).
 
-    :param weight: Weight array to compress
+    :param weight: Weight array to compress.
     :param reduction_axes: Axis or axes along which to reduce (collect) different statistics (e.g. min, max).
-    :return: compressed weights in unsigned int8, scale and zero point that was used for its quantization.
+    :param config: Information on how to compress (quantize) a specific weight.
+    :return: The compressed weights, scale and zero point that was used for its quantization.
     """
-    num_bits = 8
+    mode = config.mode
+    assert mode != CompressWeightsMode.NF4, "The function supports integer quantization only"
+    group_size = config.group_size
+    num_bits = config.num_bits
+
     level_low = 0
     level_high = 2**num_bits - 1
 
-    min_values = np.min(weight, axis=reduction_axes, keepdims=True)
-    max_values = np.max(weight, axis=reduction_axes, keepdims=True)
-
-    scale, zero_point = calculate_scale_zero_point(min_values, max_values, level_low, level_high, narrow_range=False)
+    if group_size != -1:
+        # weights are reshaped from [a1, r, a2] to [a1, r//gs, gs, a2]
+        weight, reduction_axes = _reshape_weights_for_grouped_quantization(weight, reduction_axes, group_size)
+
+    if mode in [CompressWeightsMode.INT8, CompressWeightsMode.INT4_ASYM]:
+        min_values = np.min(weight, axis=reduction_axes, keepdims=True)  # [a1, r, a2] -> [a1, 1, a2]
```

Comment:
Not blocking comment: Could you consider using nncf tensor functions instead of using numpy directly. It just simple, you need import `from nncf.experimental.tensor import functions as fns` and replace np.min to fns.min.

Reply:
At least, support for the `keepdims` is needed. 
https://github.com/openvinotoolkit/nncf/pull/2225